test/suites/osconfig/systemd-resolved.robot

```diff
@@ -129,6 +131,7 @@ Stop Systemd-Resolved
 
     Systemctl    stop    systemd-resolved
     Reboot MicroShift Host
+    Wait Until Greenboot Health Check Exited
```

Comment:
Should we just do this in "Reboot MicroShift Host"? That would mean we don't have to remember to do this 2 step process every time we want to reboot.

Reply:
We are calling reboot in a few other places. Going through those places, we do not always need to wait for greenboot. I think it's the problem in a specific test rather than a global one.

Reply:
That's surprising. What's an example of a case where we do want to reboot but do not need to wait for the system to come all the way up?

Reply:
There are different ways for the system to come up. For example, in this case we only need MicroShift service to start.
https://github.com/openshift/microshift/blob/main/test/suites/backup/backup-restore-on-reboot.robot#L27

Reply:
I think this is a bigger change that might be better handled in a separate task is we decide to implement it. 
From looking at the sources, I found a couple of places where tests need to be updated if we make such a  change.

For example:
- This case might be broken because we need to switch SSH to another IP before retrying greenboot checks
https://github.com/openshift/microshift/blob/main/test/suites/network/multi-nic.robot#L217
- This case expects a failing greenboot https://github.com/openshift/microshift/blob/main/test/resources/ostree.resource#L49

Reply:
Should I create a separate task for this work?

Reply:
OK, it sounds like we do have some cases where we would not want to wait automatically. Maybe we want 2 separate keywords, where the most obvious one does wait and another one does not? Let's leave it for now and see how things go. I don't think we need a ticket for it.